sw/device/tests/chip_power_sleep_load.c

```diff
@@ -87,6 +91,13 @@ static void wdog_irq_handler(void) {
 
   CHECK_DIF_OK(
       dif_aon_timer_irq_acknowledge(&aon_timer, kDifAonTimerIrqWdogTimerBark));
+
+  // Signal a software interrupt
+  dif_rv_plic_t rv_plic_isr;
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(plic_base_addr, &rv_plic_isr));
+  CHECK_DIF_OK(dif_rv_plic_software_irq_force(&rv_plic_isr, kPlicTarget));
```

Comment:
No, the NMI comes first and can't be masked, so we get stuck at WFI. The software interrupt is triggered in the NMI handler (but does not get handled), which causes WFI to end.

The reason for not increasing watchdog thresholds is because the race shouldn't exist in the first place. As much as possible, we should avoid relying on specific timing in tests for synchronization.

Reply:
SG, thanks for the explanation, I wasn't aware of this behavior. I was thinking the same order you described but flipped the last two :)